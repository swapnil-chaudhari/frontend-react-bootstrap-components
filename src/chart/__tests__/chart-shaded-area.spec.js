import React from 'react';
import renderShallow from 'render-shallow';
import ChartShadedArea from '../chart-shaded-area';

describe('<ChartShadedArea>', () => {
    describe('when it renders', () => {
        let component;

        const scaleFn = x => 2 * x;

        beforeAll(() => {
            component = renderShallow(
                <ChartShadedArea
                    data={ [1, 2, 3] }
                    line1Scale={ { x: scaleFn, y: scaleFn } }
                    line2Scale={ { x: scaleFn, y: scaleFn } }
                    clipPathId="url(#chartArea)"
                    fillColor="red"
                />
            ).output;
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
