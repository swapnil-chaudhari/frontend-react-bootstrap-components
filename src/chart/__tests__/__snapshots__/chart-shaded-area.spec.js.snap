// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartShadedArea> when it renders has proper markup 1`] = `
<g
  className="ChartShadedArea"
>
  <path
    clipPath="url(#chartArea)"
    d="M2,2L4,4L6,6L6,6L4,4L2,2Z"
    fill="red"
  />
</g>
`;
